### PR TITLE
[scripts] [inventory-manager] Updated inventory-manager to count total items.

### PR DESCRIPTION
**Pull request by jayRyan24**
Works with full inventory or when given a description. Uses inv list and inv search within DR's command list.

### CODE REVIEW

**Comment by MahtraDR** [comment]
Hey mate, thanks for this. Looks good. Could you please make explicit calls to common methods? We're moving away from the `includes DR*` syntax, and to explicit calls like `DRC.message`

**Comment by jayRyan24** [comment]
image

image

**Comment by jayRyan24** [comment]
> Hey mate, thanks for this. Looks good. Could you please make explicit calls to common methods? We're moving away from the `includes DR*` syntax, and to explicit calls like `DRC.message`

So, the rest of the file doesn't do that, I will just update them all then.

**Comment by jayRyan24** [comment]
Think that's all of them, thanks!

**Comment by MahtraDR** [comment]
> Think that's all of them, thanks!

Oh sweet thanks.

In which case, can you please remove `include DRC` ? and it looks good to me.

**Review by rpherbig — APPROVED** [review]

**Comment by MahtraDR** [comment]
```
>;inv save
--- Lich: inventory-manager active.
--- Lich: spellmonitor active.
--- Lich: error: undefined method `bput' for #<InventoryManager:0x0000561398e19518>
Did you mean?  put
        inventory-manager:137:in `add_current_inv'                                                                                                                                                                                            
        inventory-manager:47:in `initialize'
--- Lich: inventory-manager has exited.           
```

Some bputs are missing the `DRC.` prefix still.

File: inventory-manager.lic [hunk 1]
```diff
@@ -61,14 +67,39 @@ class InventoryManager
...
     end
   
     @item_data = OpenStruct.new(YAML.load_file(@item_db_path)).to_h
   end
 
+  def get_inv_count(desc)
+    count, closed = 0, 0
+    # Counting all items in inventory
+    if desc.nil?
+      bput('inv list', 'You have:')
```
Comment (MahtraDR) [inline inventory-manager.lic:82]:
```suggestion
      DRC.bput('inv list', 'You have:')
```

File: inventory-manager.lic [hunk 2]
```diff
@@ -61,14 +67,39 @@ class InventoryManager
...
+        break if line =~ /INVENTORY HELP/i
+
+        item = line.lstrip
+        count += 1
+        closed += 1 if item.match?(/(closed)/)
+      end
+      DRC.message "You have #{count} items, #{closed} of which are (closed) containers."
+    else
+      # Counting items matching the description, i.e., inv count pouch
+      bput("inv search #{desc}", 'You rummage')
```
Comment (MahtraDR) [inline inventory-manager.lic:93]:
```suggestion
      DRC.bput("inv search #{desc}", 'You rummage')
```

File: inventory-manager.lic [hunk 3]
```diff
@@ -78,7 +109,7 @@ class InventoryManager
 
     case bput('get my vault book', 'You get', 'What were')
```
Comment (MahtraDR) [inline inventory-manager.lic:110]:
```suggestion
    case DRC.bput('get my vault book', 'You get', 'What were')
```

**Comment by MahtraDR** [comment]
I've tagged the ones I saw, there may be more.

**Review by MahtraDR — APPROVED** [review]

**Pull request merged**